model/src/main/kotlin/config/CacheConfiguration.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package config
+
+interface CacheConfiguration {
+
+    fun validate()
```

Comment:
I believe I'd rather see this return a `Boolean` and let the caller decide whether to throw an exception or handle an error otherwise.

Reply:
@sschuberth I'm not too sure if such refactoring is worth it. In that case we'd have to store errors somewhere if the caller would rather know what's why the object is invalid.
IMO in that case since this is used only internally in ORT code it's not worth to complicate it at this point.